plugins/rbac-backend/src/providers/connect-providers.ts

```diff
@@ -276,14 +276,20 @@ export async function connectRBACProviders(
 ) {
   await Promise.all(
     providers.map(async provider => {
-      const connection = new Connection(
-        provider.getProviderName(),
-        enforcer,
-        roleMetadataStorage,
-        logger,
-        auditLogger,
-      );
-      return provider.connect(connection);
+      try {
+        const connection = new Connection(
+          provider.getProviderName(),
+          enforcer,
+          roleMetadataStorage,
+          logger,
+          auditLogger,
+        );
+        return provider.connect(connection);
+      } catch (error) {
+        throw new Error(
```

Comment:
Do we want cancel all providers, if one of them fail?

Reply:
I think it is fine for a first pass. I don't see a scenario where there would be a failure here as it should be captured [here](https://github.com/PatAKnight/backstage-plugins/blob/provide-rbac-extensions/plugins/rbac-backend-module-test/src/provider/TestProvider.ts#L90:L94). And that try catch is mainly for whenever a connection with the external provider fails. The actual connection to the provider module shouldn't fail as it is just assigning values.

Also, I haven't decided if we want the ability to have roles and permissions come from many providers. I think that it shouldn't but figured that is a discussion that we can have whenever work is started on the rbac-backend-keycloak-module.